ibl_to_nwb/updated_conversion/iblconverter.py

```diff
@@ -1,20 +1,52 @@
+"""Primary base class for all IBL converters."""
+import json
+from datetime import datetime
 from shutil import rmtree
 from typing import Optional
 
 from neuroconv import ConverterPipe
 from one.api import ONE
-from pydantic import DirectoryPath
 from pynwb import NWBFile
 
 
 class IblConverter(ConverterPipe):
-    def __init__(self, cache_folder: DirectoryPath, data_interfaces: list):
-        self.cache_folder = cache_folder
+    def __init__(self, one: ONE, data_interfaces: list):
+        self.one = one
         super().__init__(data_interfaces=data_interfaces)
 
     def get_metadata(self):
-        one = ONE(base_url="https://openalyx.internationalbrainlab.org", password="international", silent=True)
-        # TODO: fetch session and subject-level metadata, including comments/notes
+        metadata = super().get_metadata()  # Aggregates from the interfaces
+
+        session_metadata = self.one.alyx.rest(url="sessions", action="list", id=self.session)[0]
+        lab_metadata = next(lab for lab in self.one.alyx.rest("labs", "list") if lab["name"] == session_metadata["lab"])
+
+        metadata["NWBFile"]["session_id"] = f"{metadata['NWBFile']['session_start_time']}_{session_metadata['number']}"
+        metadata["NWBFile"]["identifier"] = session_metadata["id"]  # The eid is more appropriate in place of a UUID
```

Comment:
Wouldn't it be better to map their session_metadata to NWBFile.session_id and generate an id?

Reply:
So, their `session_metadata["id"]`, or `eid` as they call them, _are_ UUID's - they're not really human readable

Since the session ID is the thing that DANDI files extract and set as their file names, shouldn't it be more human readable?

From looking at past dandisets, it seems to have been convention to set our NWB file session IDs to the session start time,  and/or date of the session together with the daily session number (if running multiple sessions per day)

Those are much more human readable IMO

Then again, if someone knows in advance the `eid` from the `ibllib` that they want to load from an NWB file in the dandiset, it would be easier for them to do a lookup if the session ID matched that...

I'm just not sure how frequent of a use case that would be, because if they're using the `ibllib` they'd have access to all the source data as a live resource anyway